src/reuse/modules/mobile/device.ts

```diff
@@ -14,42 +18,342 @@ export class Device {
+      console.error("Failed to switch device orientation to landscape:", error);
+    }
+  }
+
+  /**
+   * @function switchToPortrait
+   * @memberof mobile.device
+   * @description Switches the device orientation to portrait mode.
+   * @returns {Promise<void>} Resolves when the orientation is successfully switched.
+   * @example
+   * await mobile.device.switchToPortrait();
+   */
+  async switchToPortrait(): Promise<void> {
+    const vl = this.vlf.initLog(this.switchToPortrait);
+
+    try {
+      const currentOrientation = await browser.getOrientation();
+      if (currentOrientation === "PORTRAIT") {
+        vl.log("Device is already in portrait mode.");
+        console.log("Device is already in portrait mode.");
+        return;
+      }
+
+      console.log("Switching device orientation to portrait...");
+      await browser.setOrientation("PORTRAIT");
+      console.log("Device orientation successfully switched to portrait.");
+    } catch (error) {
+      this.ErrorHandler.logException(error, `Error: Failed to switch device orientation to portrait`, true);
+      console.error("Failed to switch device orientation to portrait:", error);
+    }
+  }
+
+  /**
+   * @function getCurrentOrientation
+   * @memberof mobile.device
+   * @description Switches the device orientation to portrait mode.
+   * @returns {Promise<string>} Resolves when the orientation is successfully switched.
+   * @example
+   * await mobile.device.getCurrentOrientation();
+   */
+  async getCurrentOrientation(): Promise<string> {
+    const vl = this.vlf.initLog(this.getCurrentOrientation);
+
+    let orientation = "UNKNOWN"; // Default value
+    try {
+      orientation = await browser.getOrientation();
+      vl.log(`Current device orientation: ${orientation}`);
+      console.log(`Current device orientation: ${orientation}`);
+    } catch (error) {
+      this.ErrorHandler.logException(error, `Error: Failed to get the current device orientation`, true);
+      console.error("Failed to get the current device orientation", error);
+    }
+    return orientation;
+  }
+
+  /**
+   * @function hideKeyboard
+   * @memberof mobile.device
+   * @description Hides the keyboard on both Android and iOS using specific strategies with timeout.
+   * @param {string} strategy - Strategy to use for hiding the keyboard ('pressKey', 'tapOutside', 'swipeDown').
+   * @param {string} key - Key to press if using the 'pressKey' strategy (e.g., 'Done', 'Enter').
+   * @param {number} keyCode - Key code for Android (optional).
+   * @param {number} timeout - Timeout in milliseconds for retrying to hide the keyboard (default: 5000ms).
+   * @returns {Promise<void>}
+   */
+  async hideKeyboard(strategy: hideKeyboardStrategy, key?: string, keyCode?: number, timeout: number = 5000): Promise<void> {
+    const vl = this.vlf.initLog(this.hideKeyboard);
+
+    const startTime = Date.now();
+    while (Date.now() - startTime < timeout) {
```

Comment:
can you explain why this loop is required? Can't we use `waitUntil`?

Reply:
Sometimes the keyboard may not disappear immediately. Add retries and a timeout to handle such cases.